### DISCUSSION TITLE
Using group view as deployment view issue

### DISCUSSION THREAD

**Discussion by anatoly-kryzhanovsky** [likec4/likec4#1175] [General]
hi all
first - likec4 is amazing product with many killer features and i love it
after 1.15.0 was released i thought that i can use group view for deployment diagram, but i faced up with following problem

imagine that i have two kind of infrastructure - onprem and cloud. in each of them i have multiple cluster. now i want to visualize where and what i deploued. i have multiple application in system, and each of them deployt to multiple cluster.

so in the code it will be like this:
```
model {
    someModule1 = application 'SomeModule1' {
        somesystem1 = system 'SomeSystem1'        
    }
    someModule2 = application 'SomeModule2' {
        somesystem2 = system 'SomeSystem2'
    }
    someModule3 = application 'SomeModule3' {
        somesystem3 = system 'SomeSystem3'
    }
    
    someModule1.somesystem1 -> someModule3.somesystem3
    someModule3.somesystem3 -> someModule2.somesystem2
}

views {
    view deployment {
        group 'on-prem' {            
            color gray
            opacity 50%
            
            group 'cluster1' {
                color amber
                opacity 50%

                include somesystem1
                include somesystem2            
            }

            group 'cluster2' {
                color amber
                opacity 50%

                include somesystem1
                include somesystem2            
            }

            group 'cluster3' {
                color amber
                opacity 50%

                include somesystem1
                include somesystem2            
            }                
        }

        group 'cloud' {
            color gray
            opacity 50%

            group 'c-1' {                
                color amber
                opacity 50%

                include somesystem3
            }

            group 'c-2' {
                color amber
                opacity 50%

                include somesystem1
                include somesystem2            
            }
        }
    }
}
```

but `somesystem1` and `somesystem2` will be rendered only for cluster1 and `somesystem3` will be rendered to c-1

![image](https://github.com/user-attachments/assets/3e988574-ed0d-4221-9bbe-12d6dbf37d5a)

so i can't use that type for deployment diagram
is there are any plans or workardound to solve that problem?

**Comment by anatoly-kryzhanovsky** [comment]
no luck here?

**Comment by hubertmis** [comment]
I have the same issue.

I think it could be better to enhance the `model` syntax with the deployment information, similar to [structurizr syntax](https://docs.structurizr.com/dsl/language#deploymentenvironment).

This kind of syntax would clearly show what is deployed where and would allow controlling relationships between the deployed nodes. Moreover, it would follow the principle to model software (including deployment) in the `model {}` and describe the views (based on the model) in `views {}`.

The solution with `group`s seems a little hacky to me because it defines the software deployment model in the `views {}`.

**Reply by davydkov** [reply]
> The solution with groups seems a little hacky to me because it defines the software deployment model in the views {}.

Oh, `group` intention is to provide some instrument for visualization only. Imagine you want something like “swimlines", or box around elements with same tags.

**Reply by hubertmis** [reply]
Sure, I like the idea of groups, as long as they are used as intended - as "swimlines". When we start using them to show deployment nodes, it starts getting hacky
